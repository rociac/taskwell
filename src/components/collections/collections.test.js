import Collections from './collections.component';
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { cleanup } from '@testing-library/react';

afterEach(cleanup);

describe('Collections', () => {
  const mockProjects = [
    {
      id: 1,
      name: 'Test Project',
      image_url: 'http://www.example.com/image',
      project_type: 'Software'
    },
    {
      id: 2,
      name: 'Test Project 2',
      image_url: 'http://www.example.com/image2',
      project_type: 'Software'
    }
  ];
  it('Renders correctly', () => {
    const history = createMemoryHistory();
    const tree = renderer
      .create(
          <Router history={history}>
            <Collections projects={mockProjects} />
          </Router>
        )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});