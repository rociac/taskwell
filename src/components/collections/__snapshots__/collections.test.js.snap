// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collections Renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="container"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="project-preview"
        className="image"
      />
    </div>
    <div
      className="textContainer"
    >
      <a
        className="link"
        href="/projects/1"
        onClick={[Function]}
      >
        <p
          className="title"
        >
          Test Project
        </p>
      </a>
      <p
        className="type"
      />
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="imageContainer"
    >
      <img
        alt="project-preview"
        className="image"
      />
    </div>
    <div
      className="textContainer"
    >
      <a
        className="link"
        href="/projects/2"
        onClick={[Function]}
      >
        <p
          className="title"
        >
          Test Project 2
        </p>
      </a>
      <p
        className="type"
      />
    </div>
  </div>
</div>
`;
